indexer/cli/cli.go

```diff
@@ -32,8 +37,15 @@ func runIndexer(ctx *cli.Context) error {
 		return err
 	}
 
-	cfg.Logger = logger
-	indexer, err := indexer.NewIndexer(cfg)
+	logger = log.NewLogger(cfg.Logger)
+
+	db, err := database.NewDB(cfg.DB)
```

Comment:
Dependency injecting the DB. Doesn't make a difference but is slightly cleaner and makes the configs more explicit imo (obvious that only DB is using this config)